### ISSUE TITLE
E2 mix(vvn) implemented backwards

### ISSUE THREAD

**Issue by Denneisk** [wiremod/wire#2716]
https://github.com/wiremod/wire/blob/ccfd65ee6bf597e8d2349da78c2e13362284f6ad/lua/entities/gmod_wire_expression2/core/vector.lua#L524-L530

`mix(vvn)` is implemented so that `mix(vec1, vec2, 0) == vec2`. This is confusing and I believe a coding error.

The expected behavior is `mix(vec1, vec2, 0) == vec1`.

Not sure how no-one has complained about this before.

**Comment by Vurv78** [comment]
With https://github.com/wiremod/wire/discussions/2721, could probably fix this ordering issue with the `color` type and deprecate this function. And then maybe add `lerp(vvn)` too.

**Comment by Fasteroid** [comment]
Agreed, it's definitely backwards, and I've known for ages.  I just never complained because fixing something like this requires either deprecating functions or breaking backwards-compat (lol)

**Comment by Hazeofdream** [comment]
when e2 literally breaks chips by fixing its self lmao
